### PR TITLE
Added -ForceDelete option for Remove-AzOperationalInsightsworkspace, add cmdlet for list/restore deleted workspace

### PR DESCRIPTION
**Pull request by VeryEarly**
## Description

[design](https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/668)

## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/OperationalInsights/OperationalInsights/Az.OperationalInsights.psd1 [hunk 1]
```diff
@@ -121,7 +121,9 @@ CmdletsToExport = 'New-AzOperationalInsightsAzureActivityLogDataSource',
                'Remove-AzOperationalInsightsCluster', 
                'Get-AzOperationalInsightsLinkedService', 
                'Set-AzOperationalInsightsLinkedService', 
-               'Remove-AzOperationalInsightsLinkedService'
+               'Remove-AzOperationalInsightsLinkedService', 
+               'Get-AzOperationalInsightsDeletedWorkspace',
```
Comment (dingmeng-xue) [inline src/OperationalInsights/OperationalInsights/Az.OperationalInsights.psd1:125]: Can this cmdlet reuse Get-AzOperationalInsightsWorkspace?

**Comment by dingmeng-xue** [comment]
We need a cmdlet design review. Please create or post link on your PR.

File: src/OperationalInsights/OperationalInsights/Workspaces/RestoreAzureOperationalInsightsWorkspaceCommand.cs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The geographic region that the workspace will be created in.")]
+        [LocationCompleter("Microsoft.OperationalInsights/workspaces")]
+        [ValidateNotNullOrEmpty]
+        public string Location { get; set; }
+
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The service tier of the workspace.")]
+        [PSArgumentCompleter("free", "standard", "premium", "pernode", "standalone", "pergb2018")]
+        public string Sku { get; set; }
```
Comment (yoramsinger) [inline src/OperationalInsights/OperationalInsights/Workspaces/RestoreAzureOperationalInsightsWorkspaceCommand.cs:47]: Why did you add all those parameters to the Restore command? This should be much simple, just get the name, group and location. All the rest can be updated latter after the workspace has been restored.

Comment (VeryEarly) [reply src/OperationalInsights/OperationalInsights/Workspaces/RestoreAzureOperationalInsightsWorkspaceCommand.cs:47]: Followed description from issue:
https://github.com/Azure/azure-cli/issues/13456

"Calling PUT /workspaces to re-create the workspace. If you specify just the location and empty properties, the workspace is recovered with the same properties it had when it was deleted. You can optionally provide properties that will override the ones we keep for that workspace. The behavior is PATCH like."

File: src/OperationalInsights/OperationalInsights/Workspaces/RestoreAzureOperationalInsightsWorkspaceCommand.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        public string Location { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Don't ask for confirmation.")]
+        public SwitchParameter Force { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if (!OperationalInsightsClient.DeletedWorkspace(ResourceGroupName, Name))
+            {
+                throw new ArgumentException("workspace: " + Name + " under resource group: " + ResourceGroupName + " was not deleted");
```
Comment (yoramsinger) [inline src/OperationalInsights/OperationalInsights/Workspaces/RestoreAzureOperationalInsightsWorkspaceCommand.cs:51]: "was not deleted" change to "is not available to restore"

File: src/OperationalInsights/OperationalInsights/Workspaces/RestoreAzureOperationalInsightsWorkspaceCommand.cs (hunk 2 above)
Comment (dingmeng-xue) [reply src/OperationalInsights/OperationalInsights/Workspaces/RestoreAzureOperationalInsightsWorkspaceCommand.cs]: Behavior is a little strange that SKU can be changed during restore. I agree to Yoram.
